### PR TITLE
Fix locking annotations and add one missing lock

### PR DESCRIPTION
**Pull request by gaul**
This allows clang's thread safety checks to pass.

### CODE REVIEW

**Comment by gaul** [comment]
I don't remember or understand the state of the outstanding thread safety PRs.  I drafted this minimal one which should unblock #2493 and obsolete #2487.

**Review by ggtakec — APPROVED** [review]

**Comment by ggtakec** [comment]
> I don't remember or understand the state of the outstanding thread safety PRs. I drafted this minimal one which should unblock https://github.com/s3fs-fuse/s3fs-fuse/pull/2493 and obsolete https://github.com/s3fs-fuse/s3fs-fuse/pull/2487.

I agree with moving forward with #2493.
And I remember the contents of #2487.
After this merge, I will send a PR that will allow us to perform minimal locking, so please check it out.

**Pull request merged**
